### ISSUE TITLE
git: Doesn't pick up new tags for already installed revision

### ISSUE THREAD

**Issue by pfalcon** [ansible/ansible-modules-core#527]
If git module sees that revision to install matches that of already installed, it stop there. However, it may be the case that this revision was tagged since the previous install, and with the current behavior, git module won't fetch this new tag. This may in turn affect software which uses "git describe" command to show version information - instead the latest of tag it shows something else (raw revision or version based on previous tag).

**Comment by pfalcon** [comment]
```
$ ansible-playbook --version
ansible-playbook 1.7.1
```

**Comment by abadger** [comment]
If you can give examples of ansible playbooks or command lines that misbehave and show what hte desired outcome vs the current outcome is, that would help us look into this.  Thanks!

**Comment by cjunge-work** [comment]
I've just run into this issue but for branches. Below is the steps on how it happened for us.

Ansible 1.8.4

1) Create playbook with task that checks out master branch:

```
-   name: Clone/update repo
    git:
        dest: "{{ cache_dir }}"
        repo: "git@github.com:{{ repo }}"
        version: "master"
        update: yes
        force: yes
        recursive: yes
```

2) Add new branch to repo (eg. production), do not add new commit. Push branch to origin.

3) Run playbook again with new branch name

```
-   name: Clone/update repo
    git:
        dest: "{{ cache_dir }}"
        repo: "git@github.com:{{ repo }}"
        version: "production"
        update: yes
        force: yes
        recursive: yes
```

error: `msg: Failed to checkout production`

Doing a manual git fetch origin fixes it. I assume this would apply to tags as well.

**Comment by fredsted** [comment]
I'm running ansible 1.9.4 and new tags aren't recognised. Getting `msg: Failed to checkout tagname`.

Am I specifying the tag wrong?

**Comment by robinro** [comment]
I can't reproduce this with either 2.0 or 1.9.4.

I wrote a testcase that is the case described above in https://github.com/robinro/ansible/blob/expand_test_git/test/integration/roles/test_git/tasks/main.yml#L513-L555
It does
- checkout a repo
- make sure the _newtag_ does not exist
- add the tag to the remote repo
- update the checkout with `version=newtag`
- make sure the new repo has _newtag_

@fredsted @cjunge-work 
Please have a look at the testcase and give a similar complete set of tasks that fails for you if possible.

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by jctanner** [comment]
@pfalcon The git module and ansible have undergone many revisions since you filed this issue. Since we haven't heard from you in a while and we were waiting on further examples, I am going to close this issue. If you are able to reproduce the problem with a newer version of ansible, please file a new issue with an example playbook and -vvvv output.

**Comment by sunshine69** [comment]
Not sure why it is closed, I have hit the same problems with ansible 2.2 now

**Comment by tafax** [comment]
Same here. Ansible 2.2 seems to have the same problem.
